### PR TITLE
Make sure the iOS / tvOS versions used for running tests in CI are consistent between jobs

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
## What does this do?

Makes sure that the versions of iOS and tvOS that are used in CI are in fact those that are described in the GitHub workflows. They are currently just arbitrarily chosen since the versions requested are not in fact available on the CI runners, and we have nothing that catches this issue. So, we:

- make the job fail if the desired OS version isn't available
- change the OS versions we run our CI tests against to use ones actually available in the GitHub runners’ environment
- update the README to be truthful about which OS versions we test against.

See commit messages for more details.

## Why are we doing this?

To ensure that our tests run in a consistent environment.

## Observations

We are not (neither before nor after this PR) testing the library on most of the OS versions that we claim to support. Do we want to put effort into resolving this? @QuintinWillison?

### CODE REVIEW

**Review by QuintinWillison — APPROVED** [review]
This is excellent - solid approve from me. 👍 

In terms of testing other OS versions, we _should_ be doing this but I also think what you present here is a solid step forwards. Perhaps create an issue for that?

**Review by maratal — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -22,12 +21,20 @@ This SDK is compatible with projects that target:
...
 
-We maintain compatibility and explicitly support these platform versions, including performing CI testing on all library revisions.
+We maintain compatibility and explicitly support these platform versions.
 
-We do not explicitly maintain compatibility with older platform versions; we no longer perform CI testing on iOS 8 as of version 1.0.12 (released on January 31st 2018). Any known incompatibilities with older versions can be found [here](https://github.com/ably/ably-cocoa/issues?q=is%3Aissue+is%3Aopen+label%3A%22compatibility%22).
+We do not explicitly maintain compatibility with older platform versions. Any known incompatibilities with older versions can be found [here](https://github.com/ably/ably-cocoa/issues?q=is%3Aissue+is%3Aopen+label%3A%22compatibility%22).
 
 If you find any issues with unsupported platform versions, please [raise an issue](https://github.com/ably/ably-cocoa/issues) in this repository or [contact Ably customer support](https://support.ably.io) for advice.
 
+### Continuous integration testing
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply README.md]: Updated 👍

**Comment by lawrence-forooghian** [comment]
> Perhaps create an issue for that?

Created #1292.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply README.md]: This is a minor one and I'm keen to merge to unblock further PRs, so I'm resolving this one, @QuintinWillison.

**Pull request merged**